main.go

```diff
@@ -396,6 +486,177 @@ func runInPod(name string, cmdToRun string, env []string, timeout int) ([]string
 	return lines[:len(lines)-1], timeout_reached
 }
 
+func selectNodes(step *Step, config Config, subsetPartition map[int][]int) []int{
+	/* Handle OnNode EndNode syntax */
+	if step.OnNode > 0 {
+		if step.Selection != nil {
+			fatal("Two node selection methods on test step")
+		}
+		if step.EndNode == 0 {
+			step.EndNode = step.OnNode
+			selected := make([]int, 1)
+			selected[0] = step.OnNode
+			return selected
+		}
+		return makeRange(step.OnNode, step.EndNode)
+	} else if step.Selection == nil {
+		fatal("No selection method on test step")
```

Comment:
I like this better too.  One unfortunate thing is that here we can't switch on step so I am going with the slightly if else variant of this.